src/main/java/com/softserve/academy/spaced/repetition/service/impl/DeckServiceImpl.java

```diff
@@ -46,6 +41,12 @@
     @Autowired
     private FolderService folderService;
 
+    @Autowired
+    private DeckPriceRepository deckPriceRepository;
```

Comment:
Now that you have CascadeType.ALL, please try to remove deckPriceRepository. It should work without.

- when you save a Deck, its DeckPrice should be saved according CascadeType.PERSIST.
- when you delete a Deck, its DeckPrice should be deleted according to CascadeType.DELETE

If that is the case please also remove DeckPriceRepository class.